src/libpmemfile/preload.c

```diff
@@ -225,19 +226,182 @@ exit_with_msg(int ret, const char *msg)
 	__builtin_unreachable();
 }
 
+static size_t page_size;
+
+#ifdef VALIDATE_POINTERS
+
+static bool validate_pointers;
+
+static void
+initialize_validate_pointers(void)
+{
+	const char *env_str = getenv("PMEMFILE_PRELOAD_VALIDATE_POINTERS");
+	if (env_str)
+		validate_pointers = env_str[0] == '1';
+}
+
+/*
+ * returns true when [buf, buf+len] is accessible
+ */
+static bool
+is_accessible(const void *ptr, size_t len)
+{
+	if (!validate_pointers)
+		return true;
+	if (!ptr || len == 0)
+		return true;
```

Comment:
So, if `ptr` is NULL, we say it is accessible????several question marks???

Reply:
The code is correct, but naming is not. This function is supposed to say whether memory is safe to handle. NULL is not accessible, but all APIs should be prepared to handle this one specific value. Ideas for better name?

is_safe? is_valid? is_poisoned?

Reply:
I moved this check out of is_accessible.